python/cuml/preprocessing/LabelEncoder.py

```diff
@@ -180,7 +180,7 @@ def fit(self, y, _classes=None):
         if _classes is not None:
             self.classes_ = _classes
         else:
-            self.classes_ = y.unique().sort_values(
+            self.classes_ = y.drop_duplicates().sort_values(
                 ignore_index=True
             )  # dedupe and sort
```

Comment:
Do you see a reason why we are not just calling this within the `if _classes is None:` block on line 174? I think that would simplify the logic and makes it much clearer that `y` must be a data frame at that point. Or could the `y.dtype` change through this operation?

Reply:
Nope, that looks like a pretty safe change to me. I'll reshuffle that a bit.